Bluetooth: audio: Add possibility to use static broadcast id (#80228)
zephyrproject-rtos/zephyr: doc/releases/migration-guide-4.0.rst

  do a search-and-replace for ``bt_audio_codec_qos`` to ``bt_bap_qos_cfg`` and
  ``BT_AUDIO_CODEC_QOS`` to ``BT_BAP_QOS_CFG``. (:github:`76633`)

> * The generation of broadcast code inside of zephyr stack has been removed, it is now up the
>   application to generate a broadcast id. This means that the application can now fully decide
>   whether to use a static or random broadcast id. Reusing and statically defining a broadcast ID was
>   added to the Basic Audio Profile in version 1.0.2, which is the basis for this change. All
>   instances of :c:func:`bt_cap_initiator_broadcast_get_id` and
>   :c:func:`bt_bap_broadcast_source_get_id` has been removed(:github:`80228`)

Reviewer: ```suggestion
* The generation of broadcast ID inside of zephyr stack has been removed, it is now up the
  application to generate a broadcast ID. This means that the application can now fully decide
  whether to use a static or random broadcast ID. Reusing and statically defining a broadcast ID was
  added to the Basic Audio Profile in version 1.0.2, which is the basis for this change. All
  instances of :c:func:`bt_cap_initiator_broadcast_get_id` and
  :c:func:`bt_bap_broadcast_source_get_id` has been removed(:github:`80228`)
```

Missed a few IDs :)

  do a search-and-replace for ``bt_audio_codec_qos`` to ``bt_bap_qos_cfg`` and
  ``BT_AUDIO_CODEC_QOS`` to ``BT_BAP_QOS_CFG``. (:github:`76633`)

> 
> * The generation of broadcast code inside of zephyr stack has been removed, it is now up the

Reviewer: ```suggestion
* The generation of broadcast code inside of zephyr stack has been removed, it is now up the
```

Reviewer: One too many empty lines :)

> * The generation of broadcast code inside of zephyr stack has been removed, it is now up the
>   application to generate a broadcast id. This means that the application can now fully decide
>   whether to use a static or random broadcast id. All instances of
>   :c:func:`bt_cap_initiator_broadcast_get_id` and :c:func:`bt_bap_broadcast_source_get_id` has
>   been removed, and to migrate these calls need to be removed. The provided uint32_t to these
>   function calls should now be set by the application instead, either to a random value, or
>   a predefined static value. (:github:`80228`)

Reviewer: ```suggestion
* The generation of broadcast ID inside of Zephyr stack has been removed, it is now up the
  application to generate a broadcast ID. This means that the application can now fully decide
  whether to use a static or random broadcast ID. Reusing and statically defining a broadcast ID was added to the Basic Audio Profile in version 1.0.2. All instances of
  :c:func:`bt_cap_initiator_broadcast_get_id` and :c:func:`bt_bap_broadcast_source_get_id` has
  been removed, and to migrate these calls need to be removed. (:github:`80228`)
```

`id` should be `ID`. 

Removed a sentence as it was confusing, and probably shouldn't refer to a ``uint32_t`` for a 24-bit value to set in the advertising data :) 

Also added a reference to BAP 1.0.2 that made this legal behavior

Author: Oh yes, nice catch. Really shouldn't have been a 32-bit mentioned at all.
